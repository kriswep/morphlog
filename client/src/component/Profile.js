import React from 'react';
import gql from 'graphql-tag';
import { graphql, compose } from 'react-apollo';

class Profile extends React.Component {
  state = {
    email: '',
    password: '',
  };

  dispatch = (e, v, x) => {
    if (e.target.dataset && e.target.dataset.state) {
      const newState = {};
      newState[e.target.dataset.state] = e.target.value;
      this.setState(newState);
    }
  };

  signup = e => {
    this.props
      .signupMutation({
        variables: {
          email: this.state.email,
          password: this.state.password,
        },
      })
      .then(({ data }) => {
        console.log('got data', data);
        localStorage.setItem('auth', data.signup.token);
      })
      .catch(error => {
        console.log('there was an error sending the query', error);
      });
  };

  signin = e => {
    this.props
      .signinMutation({
        variables: {
          email: this.state.email,
          password: this.state.password,
        },
      })
      .then(({ data }) => {
        console.log('got data', data);
        localStorage.setItem('auth', data.login.token);
      })
      .catch(error => {
        console.log('there was an error sending the query', error);
      });
  };

  render() {
    console.log(this.props.profileQuery);
    return (
      <div>
        <h2>Profile</h2>
        <input
          type="text"
          placeholder="email"
          data-state="email"
          onChange={this.dispatch}
        />
        <input
          type="password"
          placeholder="password"
          data-state="password"
          onChange={this.dispatch}
        />
        <button onClick={this.signup}>SignUp</button>
        <button onClick={this.signin}>SignIn</button>
        {!this.props.profileQuery.loading && (
          <section>
            <p>Name: {this.props.profileQuery.me.name}</p>
            <p>E-Mail: {this.props.profileQuery.me.email}</p>
          </section>
        )}
      </div>
    );
  }
}

const SIGNUP_MUTATION = gql`
  mutation s($email: String!, $password: String!) {
    signup(email: $email, password: $password, name: "demo3") {
      token
      user {
        id
        name
        email
      }
    }
  }
`;
const SIGNIN_MUTATION = gql`
  mutation signin($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
      user {
        id
        name
        email
      }
    }
  }
`;
const PROFILE_QUERY = gql`
  query profile {
    me {
      id
      name
      email
    }
  }
`;

export default compose(
  graphql(PROFILE_QUERY, { name: 'profileQuery' }),
  graphql(SIGNUP_MUTATION, { name: 'signupMutation' }),
  graphql(SIGNIN_MUTATION, { name: 'signinMutation' }),
)(Profile);
