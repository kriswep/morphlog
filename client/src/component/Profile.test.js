/* globals test expect jest */
import React from 'react';
import { ApolloProvider } from 'react-apollo';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import createClient from '../../utils/apolloMocks';

import Profile from './Profile';

const mocks = {
  Query: () => ({
    me: () => ({
      id: 'cjcp94wxp025801100npb28yg',
      email: 'email',
      name: 'name',
      createdAt: '2018-01-21T20:45:06.000Z',
      updatedAt: '2018-01-21T20:45:06.000Z',
    }),
  }),
};
const client = createClient(mocks);

test('Profile renders correctly', async () => {
  const wrapper = mount(
    <ApolloProvider client={client}>
      <Profile />
    </ApolloProvider>,
  );
  await new Promise(res => window.setTimeout(res, 1));
  wrapper.setProps({ foo: 'bar' }); // poke it to rerender...
  // todo: think mock apollo error
  // expect(toJson(wrapper.find('[data-test="authenticate"]'))).toMatchSnapshot();
  await new Promise(res => window.setTimeout(res, 1));
  wrapper.setProps({ projectId: 'bar' }); // poke it to rerender...
  expect(toJson(wrapper.find('[data-test="profile"]'))).toMatchSnapshot();
});
