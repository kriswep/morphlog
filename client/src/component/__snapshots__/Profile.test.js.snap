// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
Array [
  <Message
    data-test="profile"
>
    <div
        className="ui message"
        data-test="profile"
    >
        <p>
            Name: 
            name
        </p>
        <p>
            E-Mail: 
            email
        </p>
        <Button
            as="button"
            color="teal"
            fluid={true}
            onClick={[Function]}
            size="large"
        >
            <button
                className="ui teal large fluid button"
                onClick={[Function]}
                role="button"
            >
                LogOut
            </button>
        </Button>
    </div>
</Message>,
  <div
    className="ui message"
    data-test="profile"
>
    <p>
        Name: 
        name
    </p>
    <p>
        E-Mail: 
        email
    </p>
    <Button
        as="button"
        color="teal"
        fluid={true}
        onClick={[Function]}
        size="large"
    >
        <button
            className="ui teal large fluid button"
            onClick={[Function]}
            role="button"
        >
            LogOut
        </button>
    </Button>
</div>,
]
`;
